internal/services/compute/orchestrated_virtual_machine_scale_set_resource.go

```diff
@@ -203,7 +994,46 @@ func resourceOrchestratedVirtualMachineScaleSetDelete(d *pluginsdk.ResourceData,
 		return err
 	}
 
-	// @ArcturusZhang (mimicking from linux_virtual_machine_pluginsdk.go): sending `nil` here omits this value from being sent
+	// Upgrading to the 2021-07-01 exposed a new expand parameter in the GET method
+	resp, err := client.Get(ctx, id.ResourceGroup, id.Name, compute.ExpandTypesForGetVMScaleSetsUserData)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			return nil
+		}
```

Comment:
this indicates a bug (since this should have been picked up in the Plan via calling the Read function) - so can we remove this?

Reply:
I don't think so as this is another workaround for a known bug in the API. Before we can delete the VMSS we have to scale down the capacity to 0 before we call delete else we get the pesky `subnet in use` error again. I am unaware of a better workaround at the moment. 🙁